doc/whats_new/v0.23.rst

```diff
@@ -542,6 +542,11 @@ Miscellaneous
   <https://scikit-learn-enhancement-proposals.readthedocs.io/en/latest/slep009/proposal.html>`_
   for more details.
 
+- |API| The default setting `print_changed_only` has been changed from False
+  to True. This means that the `repr` of estimators is now more concise and
+  only shows the parameters whose default value has been changed when
+  printing an estimator. :pr:`17061` by `Nicolas Hug`_.
```

Comment:
Maybe we should include the snippet to restore the previous behavior in the whats new.

Maybe we can also mention that it's always possible to quickly introspect the default values of an estimator by using the `get_params()` method in an interactive jupyter notebook or ipython session:

```python
from sklearn.ensemble import RandomForestClassifier

RandomForestClassifier().get_params()
```